src/aliceVision/sfm/utils/alignment.cpp

```diff
@@ -1026,6 +1026,62 @@ void computeCentersVarCov(const sfmData::SfMData& sfmData, const Vec3 & mean, Ei
     }
 }
 
+void computeNewCoordinateSystemGroundAuto(const sfmData::SfMData& sfmData, Vec3& out_t)
+{
+    out_t.fill(0.0);
+
+    // Collect landmark positions for ground detection
+    // Note: the Y axis is pointing down, therefore +Y is the direction to the ground
+    std::vector<Vec3> points;
+    for (auto & plandmark: sfmData.getLandmarks())
+    {
+        // Filter out landmarks with not enough observations
+        if (plandmark.second.observations.size() < 3)
+        {
+            continue;
+        }
+
+        // Filter out landmarks that lie above all the cameras that observe them
+        // This filtering step assumes that cameras should not be underneath the ground level
+        const Vec3 X = plandmark.second.X;
+        bool foundUnder = false;
+        for (const auto & pObs : plandmark.second.observations)
+        {
+            const IndexT viewId = pObs.first;
+            const Vec3 camCenter = sfmData.getPose(sfmData.getView(viewId)).getTransform().center();
+
+            if (X(1) > camCenter(1))
+            {
+                foundUnder = true;
+                break;
+            }
+        }
```

Comment:
Why not just look globally for the lowest camera and filter out all the points above?

And to avoid a problem with one failed solved cameras far away underground, check the bbox of landmarks and bbox of camera centers, to ignore cameras below the landmarks' bbox.